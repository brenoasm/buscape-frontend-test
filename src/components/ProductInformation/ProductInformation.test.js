import React from "react";
import { shallow } from "enzyme";

import ProductInformation from "./";

describe("ProductInformation", () => {
  const product = {
    id: 2321312,
    name: "Smartphone Apple iPhone 7 128GB",
    images: [
      {
        url:
          "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
        selected: true
      },
      {
        url:
          "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
        selected: false
      },
      {
        url:
          "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
        selected: false
      },
      {
        url:
          "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
        selected: false
      }
    ],
    price: {
      value: 3509.1,
      installments: 10,
      installmentValue: 389.9
    },
    selectedImage:
      "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
    loading: false
  };

  it("matches the snapshot", () => {
    const wrapper = shallow(<ProductInformation product={product} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should contain a FavoriteButton Component", () => {
    const wrapper = shallow(<ProductInformation product={product} />);

    expect(wrapper.find('FavoriteButton')).toHaveLength(1);
  });

  it("should contain a AddToCartButton Component", () => {
    const wrapper = shallow(<ProductInformation product={product} />);

    expect(wrapper.find('AddToCartButton')).toHaveLength(1);
  });

  it("should contain a Loading Component", () => {
    const loadingProduct = {
      id: 2321312,
      name: "Smartphone Apple iPhone 7 128GB",
      images: [
        {
          url:
            "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
          selected: true
        },
        {
          url:
            "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
          selected: false
        },
        {
          url:
            "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
          selected: false
        },
        {
          url:
            "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
          selected: false
        }
      ],
      price: {
        value: 3509.1,
        installments: 10,
        installmentValue: 389.9
      },
      selectedImage:
        "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
      loading: true
    };

    const wrapper = shallow(<ProductInformation product={loadingProduct} />);

    expect(wrapper.find('Styled(Loading)')).toHaveLength(1);
  });

});
