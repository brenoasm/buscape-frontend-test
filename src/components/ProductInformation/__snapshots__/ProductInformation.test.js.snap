// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductInformation matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductInformation
    handleProductAddToCartClick={[Function]}
    product={
      Object {
        "id": 2321312,
        "images": Array [
          Object {
            "selected": true,
            "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
          },
          Object {
            "selected": false,
            "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
          },
          Object {
            "selected": false,
            "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
          },
          Object {
            "selected": false,
            "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
          },
        ],
        "loading": false,
        "name": "Smartphone Apple iPhone 7 128GB",
        "price": Object {
          "installmentValue": 389.9,
          "installments": 10,
          "value": 3509.1,
        },
        "selectedImage": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef>
        <div>
          <span>
            Smartphone Apple iPhone 7 128GB
          </span>
          <FavoriteButton />
        </div>
        <div>
          <div>
            <Tag
              text="MELHOR PREÇO"
            />
          </div>
          <div>
            <ProductSubtotal
              price={
                Object {
                  "installmentValue": 389.9,
                  "installments": 10,
                  "value": 3509.1,
                }
              }
            />
            <div>
              <AddToCartButton
                handleProductAddToCartClick={[Function]}
                product={
                  Object {
                    "id": 2321312,
                    "images": Array [
                      Object {
                        "selected": true,
                        "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                      },
                      Object {
                        "selected": false,
                        "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
                      },
                      Object {
                        "selected": false,
                        "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
                      },
                      Object {
                        "selected": false,
                        "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
                      },
                    ],
                    "loading": false,
                    "name": "Smartphone Apple iPhone 7 128GB",
                    "price": Object {
                      "installmentValue": 389.9,
                      "installments": 10,
                      "value": 3509.1,
                    },
                    "selectedImage": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                  }
                }
              />
            </div>
          </div>
        </div>
      </ForwardRef>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div>
            <span>
              Smartphone Apple iPhone 7 128GB
            </span>
            <FavoriteButton />
          </div>,
          <div>
            <div>
              <Tag
                text="MELHOR PREÇO"
              />
            </div>
            <div>
              <ProductSubtotal
                price={
                  Object {
                    "installmentValue": 389.9,
                    "installments": 10,
                    "value": 3509.1,
                  }
                }
              />
              <div>
                <AddToCartButton
                  handleProductAddToCartClick={[Function]}
                  product={
                    Object {
                      "id": 2321312,
                      "images": Array [
                        Object {
                          "selected": true,
                          "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                        },
                        Object {
                          "selected": false,
                          "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
                        },
                        Object {
                          "selected": false,
                          "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
                        },
                        Object {
                          "selected": false,
                          "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
                        },
                      ],
                      "loading": false,
                      "name": "Smartphone Apple iPhone 7 128GB",
                      "price": Object {
                        "installmentValue": 389.9,
                        "installments": 10,
                        "value": 3509.1,
                      },
                      "selectedImage": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                    }
                  }
                />
              </div>
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                Smartphone Apple iPhone 7 128GB
              </span>,
              <FavoriteButton />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Smartphone Apple iPhone 7 128GB",
              },
              "ref": null,
              "rendered": "Smartphone Apple iPhone 7 128GB",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <Tag
                  text="MELHOR PREÇO"
                />
              </div>,
              <div>
                <ProductSubtotal
                  price={
                    Object {
                      "installmentValue": 389.9,
                      "installments": 10,
                      "value": 3509.1,
                    }
                  }
                />
                <div>
                  <AddToCartButton
                    handleProductAddToCartClick={[Function]}
                    product={
                      Object {
                        "id": 2321312,
                        "images": Array [
                          Object {
                            "selected": true,
                            "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                          },
                          Object {
                            "selected": false,
                            "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
                          },
                          Object {
                            "selected": false,
                            "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
                          },
                          Object {
                            "selected": false,
                            "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
                          },
                        ],
                        "loading": false,
                        "name": "Smartphone Apple iPhone 7 128GB",
                        "price": Object {
                          "installmentValue": 389.9,
                          "installments": 10,
                          "value": 3509.1,
                        },
                        "selectedImage": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                      }
                    }
                  />
                </div>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Tag
                  text="MELHOR PREÇO"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "text": "MELHOR PREÇO",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <ProductSubtotal
                    price={
                      Object {
                        "installmentValue": 389.9,
                        "installments": 10,
                        "value": 3509.1,
                      }
                    }
                  />,
                  <div>
                    <AddToCartButton
                      handleProductAddToCartClick={[Function]}
                      product={
                        Object {
                          "id": 2321312,
                          "images": Array [
                            Object {
                              "selected": true,
                              "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                            },
                            Object {
                              "selected": false,
                              "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
                            },
                            Object {
                              "selected": false,
                              "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
                            },
                            Object {
                              "selected": false,
                              "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
                            },
                          ],
                          "loading": false,
                          "name": "Smartphone Apple iPhone 7 128GB",
                          "price": Object {
                            "installmentValue": 389.9,
                            "installments": 10,
                            "value": 3509.1,
                          },
                          "selectedImage": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                        }
                      }
                    />
                  </div>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "price": Object {
                      "installmentValue": 389.9,
                      "installments": 10,
                      "value": 3509.1,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <AddToCartButton
                      handleProductAddToCartClick={[Function]}
                      product={
                        Object {
                          "id": 2321312,
                          "images": Array [
                            Object {
                              "selected": true,
                              "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                            },
                            Object {
                              "selected": false,
                              "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
                            },
                            Object {
                              "selected": false,
                              "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
                            },
                            Object {
                              "selected": false,
                              "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
                            },
                          ],
                          "loading": false,
                          "name": "Smartphone Apple iPhone 7 128GB",
                          "price": Object {
                            "installmentValue": 389.9,
                            "installments": 10,
                            "value": 3509.1,
                          },
                          "selectedImage": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                        }
                      }
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "handleProductAddToCartClick": [Function],
                      "product": Object {
                        "id": 2321312,
                        "images": Array [
                          Object {
                            "selected": true,
                            "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                          },
                          Object {
                            "selected": false,
                            "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
                          },
                          Object {
                            "selected": false,
                            "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
                          },
                          Object {
                            "selected": false,
                            "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
                          },
                        ],
                        "loading": false,
                        "name": "Smartphone Apple iPhone 7 128GB",
                        "price": Object {
                          "installmentValue": 389.9,
                          "installments": 10,
                          "value": 3509.1,
                        },
                        "selectedImage": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-EHOje",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  flex-direction: column;

  > div:first-child {
    margin-bottom: 10px;

    > span:first-child {
      font-size: 26px;
      font-weight: 600;
      margin-right: 20px;
    }

    > span:last-child {
      display: inline-block;
      line-height: normal;
    }
  }

  > div:nth-child(2) {
    border-top: 1px solid ",
            "#F3F3F3",
            ";
    padding: 15px 0;

    > div:first-child {
      width: 100%;
      margin-bottom: 15px;
    }

    > div:last-child {
      display: flex;
      line-height: 35px;

      > div:last-child {
        margin-left: auto;
      }
    }
  }
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-EHOje",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef>
          <div>
            <span>
              Smartphone Apple iPhone 7 128GB
            </span>
            <FavoriteButton />
          </div>
          <div>
            <div>
              <Tag
                text="MELHOR PREÇO"
              />
            </div>
            <div>
              <ProductSubtotal
                price={
                  Object {
                    "installmentValue": 389.9,
                    "installments": 10,
                    "value": 3509.1,
                  }
                }
              />
              <div>
                <AddToCartButton
                  handleProductAddToCartClick={[Function]}
                  product={
                    Object {
                      "id": 2321312,
                      "images": Array [
                        Object {
                          "selected": true,
                          "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                        },
                        Object {
                          "selected": false,
                          "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
                        },
                        Object {
                          "selected": false,
                          "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
                        },
                        Object {
                          "selected": false,
                          "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
                        },
                      ],
                      "loading": false,
                      "name": "Smartphone Apple iPhone 7 128GB",
                      "price": Object {
                        "installmentValue": 389.9,
                        "installments": 10,
                        "value": 3509.1,
                      },
                      "selectedImage": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                    }
                  }
                />
              </div>
            </div>
          </div>
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <div>
              <span>
                Smartphone Apple iPhone 7 128GB
              </span>
              <FavoriteButton />
            </div>,
            <div>
              <div>
                <Tag
                  text="MELHOR PREÇO"
                />
              </div>
              <div>
                <ProductSubtotal
                  price={
                    Object {
                      "installmentValue": 389.9,
                      "installments": 10,
                      "value": 3509.1,
                    }
                  }
                />
                <div>
                  <AddToCartButton
                    handleProductAddToCartClick={[Function]}
                    product={
                      Object {
                        "id": 2321312,
                        "images": Array [
                          Object {
                            "selected": true,
                            "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                          },
                          Object {
                            "selected": false,
                            "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
                          },
                          Object {
                            "selected": false,
                            "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
                          },
                          Object {
                            "selected": false,
                            "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
                          },
                        ],
                        "loading": false,
                        "name": "Smartphone Apple iPhone 7 128GB",
                        "price": Object {
                          "installmentValue": 389.9,
                          "installments": 10,
                          "value": 3509.1,
                        },
                        "selectedImage": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                      }
                    }
                  />
                </div>
              </div>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  Smartphone Apple iPhone 7 128GB
                </span>,
                <FavoriteButton />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Smartphone Apple iPhone 7 128GB",
                },
                "ref": null,
                "rendered": "Smartphone Apple iPhone 7 128GB",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  <Tag
                    text="MELHOR PREÇO"
                  />
                </div>,
                <div>
                  <ProductSubtotal
                    price={
                      Object {
                        "installmentValue": 389.9,
                        "installments": 10,
                        "value": 3509.1,
                      }
                    }
                  />
                  <div>
                    <AddToCartButton
                      handleProductAddToCartClick={[Function]}
                      product={
                        Object {
                          "id": 2321312,
                          "images": Array [
                            Object {
                              "selected": true,
                              "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                            },
                            Object {
                              "selected": false,
                              "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
                            },
                            Object {
                              "selected": false,
                              "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
                            },
                            Object {
                              "selected": false,
                              "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
                            },
                          ],
                          "loading": false,
                          "name": "Smartphone Apple iPhone 7 128GB",
                          "price": Object {
                            "installmentValue": 389.9,
                            "installments": 10,
                            "value": 3509.1,
                          },
                          "selectedImage": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                        }
                      }
                    />
                  </div>
                </div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Tag
                    text="MELHOR PREÇO"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "text": "MELHOR PREÇO",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <ProductSubtotal
                      price={
                        Object {
                          "installmentValue": 389.9,
                          "installments": 10,
                          "value": 3509.1,
                        }
                      }
                    />,
                    <div>
                      <AddToCartButton
                        handleProductAddToCartClick={[Function]}
                        product={
                          Object {
                            "id": 2321312,
                            "images": Array [
                              Object {
                                "selected": true,
                                "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                              },
                              Object {
                                "selected": false,
                                "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
                              },
                              Object {
                                "selected": false,
                                "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
                              },
                              Object {
                                "selected": false,
                                "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
                              },
                            ],
                            "loading": false,
                            "name": "Smartphone Apple iPhone 7 128GB",
                            "price": Object {
                              "installmentValue": 389.9,
                              "installments": 10,
                              "value": 3509.1,
                            },
                            "selectedImage": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                          }
                        }
                      />
                    </div>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "price": Object {
                        "installmentValue": 389.9,
                        "installments": 10,
                        "value": 3509.1,
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <AddToCartButton
                        handleProductAddToCartClick={[Function]}
                        product={
                          Object {
                            "id": 2321312,
                            "images": Array [
                              Object {
                                "selected": true,
                                "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                              },
                              Object {
                                "selected": false,
                                "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
                              },
                              Object {
                                "selected": false,
                                "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
                              },
                              Object {
                                "selected": false,
                                "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
                              },
                            ],
                            "loading": false,
                            "name": "Smartphone Apple iPhone 7 128GB",
                            "price": Object {
                              "installmentValue": 389.9,
                              "installments": 10,
                              "value": 3509.1,
                            },
                            "selectedImage": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                          }
                        }
                      />,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "handleProductAddToCartClick": [Function],
                        "product": Object {
                          "id": 2321312,
                          "images": Array [
                            Object {
                              "selected": true,
                              "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                            },
                            Object {
                              "selected": false,
                              "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
                            },
                            Object {
                              "selected": false,
                              "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
                            },
                            Object {
                              "selected": false,
                              "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
                            },
                          ],
                          "loading": false,
                          "name": "Smartphone Apple iPhone 7 128GB",
                          "price": Object {
                            "installmentValue": 389.9,
                            "installments": 10,
                            "value": 3509.1,
                          },
                          "selectedImage": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-EHOje",
            "isStatic": true,
            "rules": Array [
              "
  display: flex;
  flex-direction: column;

  > div:first-child {
    margin-bottom: 10px;

    > span:first-child {
      font-size: 26px;
      font-weight: 600;
      margin-right: 20px;
    }

    > span:last-child {
      display: inline-block;
      line-height: normal;
    }
  }

  > div:nth-child(2) {
    border-top: 1px solid ",
              "#F3F3F3",
              ";
    padding: 15px 0;

    > div:first-child {
      width: 100%;
      margin-bottom: 15px;
    }

    > div:last-child {
      display: flex;
      line-height: 35px;

      > div:last-child {
        margin-left: auto;
      }
    }
  }
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-EHOje",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
